classes/ActionScheduler_AdminView.php

```diff
@@ -35,9 +35,53 @@ public function init() {
 			} else {
 				add_action( 'admin_menu', array( $self, 'register_menu' ) );
 			}
+			add_action( 'admin_notices', array( $self, 'check_past_due_actions' ) );
 		}
 	}
 
+	/**
+	 * Check if there are any past due actions.
+	 *
+	 * This method will check if there are past due actions. Because Action Scheduler is action agnostic,
+	 * it will trigger an action with a list of hooks that callbacks can use to handle past due actions
+	 * as needed.
+	 *
+	 * To narrow down possible results, and improve query looking for past due actions, any plugin
+	 * must add their hook name to an array through the `action_scheduler_past_due_hooks`
+	 * filter.
+	 *
+	 * The action owner should be listening to the `action_schedule_past_due_action` action, 
+	 * and do something to warn the users about how to fix their WP-Cron or to run their actions manually.
+	 *
+	 * This method is executed on the `admin_notices`, so it is safe to render the admin notification.
+	 */
+	public function check_past_due_actions() {
+		$instance = ActionScheduler_Store::instance();
+
+		$actions = apply_filters( 'action_scheduler_past_due_hooks', array() );
+		
+		if ( ! empty( $actions ) ) {
+			return;
+		}
+
+		$action_ids = $instance->query_actions( array(
+			'date'   => new Datetime( apply_filters( 'action_scheduler_past_due_time', '-5 days' ) ),
+			'status' => ActionScheduler_Store::STATUS_PENDING,
+			'hook'   => $actions,
+		) );
+
+		if ( ! empty( $action_ids ) ) {
+			return;
+		}
+
+		$actions = array();
+		foreach ( $action_ids as $id ) {
+			$action = $instance->fetch_action( $id );
+			$actions[ $action->get_hook() ][] = $action;
```

Comment:
Is it necessary to assign the result of `fetch_action( $id )` to `$action`? Why not compress these two lines into one: `$actions[ $action->get_hook() ][] = $instance->fetch_action()` (or as suggested above, use  `ActionScheduler::store()-> fetch_action()` instead of `$instance`).

Reply:
@thenbrent It looks like we would still need the `$action` variable. Here the code is accessing a method of the variable to use as the array key (`$action->get_hook()`), and the action object is stored to the array as well.

I still agree with using `ActionScheduler_Store::instance()->fetch()` directly.

Reply:
> Is it necessary to assign the result of fetch_action( $id ) to $action? Why not compress these two lines into one: $actions[ $action->get_hook() ][] = $instance->fetch_action() (or as suggested above, use ActionScheduler::store()-> fetch_action() instead of $instance).
 
It makes way easier for callee functions to know if there is something they care or not. Just by simply checking if a key exists or not.